pkg/fleets/controller.go

```diff
@@ -461,18 +477,53 @@ func (c *Controller) rollingUpdateActive(fleet *agonesv1.Fleet, active *agonesv1
 }
 
 // rollingUpdateRest applies the rolling update to the inactive GameServerSets
-func (c *Controller) rollingUpdateRest(fleet *agonesv1.Fleet, rest []*agonesv1.GameServerSet) error {
+func (c *Controller) rollingUpdateRest(fleet *agonesv1.Fleet, active *agonesv1.GameServerSet, rest []*agonesv1.GameServerSet) error {
 	if len(rest) == 0 {
 		return nil
 	}
 
-	r, err := intstr.GetValueFromIntOrPercent(fleet.Spec.Strategy.RollingUpdate.MaxUnavailable, int(fleet.Spec.Replicas), true)
+	// Look at Kubernetes Deployment util ResolveFenceposts() function
+	var r int
+	var err error
+	if runtime.FeatureEnabled(runtime.FeatureRollingUpdateOnReady) {
+		r, err = intstr.GetValueFromIntOrPercent(fleet.Spec.Strategy.RollingUpdate.MaxUnavailable, int(fleet.Spec.Replicas), false)
+		if r == 0 {
+			r = 1
+		}
+		if r > int(fleet.Spec.Replicas) {
+			r = int(fleet.Spec.Replicas)
+		}
+	} else {
+		r, err = intstr.GetValueFromIntOrPercent(fleet.Spec.Strategy.RollingUpdate.MaxUnavailable, int(fleet.Spec.Replicas), true)
+	}
+
 	if err != nil {
 		return errors.Wrapf(err, "error calculating scaling gameserverset: %s", fleet.ObjectMeta.Name)
 	}
 	unavailable := int32(r)
 
+	totalScaledDown := int32(0)
```

Comment:
What if we consider a name like "totalAlreadyScaledDown"? Or something similar to show more difference between this variable and totalScaleDownCount

Reply:
Agree, a bit misleading, I think it worth fixing. Please note that I have taken naming from Kubernetes original `RollingUpdate` code.